generators/entity-client/templates/vue/src/test/javascript/e2e/entities/entity-page-object.ts.ejs

```diff
@@ -35,18 +37,21 @@ export default class <%= entityClass %>ComponentsPage extends AlertPage {
   getDetailsButton(record: ElementFinder) {
     return record.element(by.css('.btn-info.details'));
   }
-
+<%_ if (!readOnly) { _%>
   getEditButton(record: ElementFinder) {
     return record.element(by.css('.btn-primary.edit'));
   }
 
   getDeleteButton(record: ElementFinder) {
     return record.element(by.css('div table .btn-danger'));
   }
+<%_ } _%>
 }
 
 export class <%= entityClass %>DeleteDialog {
```

Comment:
I think the whole class can be skipped, since the DeleteDialog should not exist for a readonly entity.